lib/rdoc/generator/template/darkfish/js/darkfish.js

```diff
@@ -106,9 +106,22 @@ function hookSidebar() {
   }
 }
 
+function hookAnchorLinks() {
+  document.addEventListener('click', (e) => {
```

Comment:
While this works well, is it possible to only add this listener to links under `#navigation`? I'm not familiar with JS but adding a global listener to click doesn't feel like the best practice 🤔

Reply:
I can add the `#navigation` checking to the selection easily if we want to avoid triggering if main content is selected.

I chose this approach to add the document event listener because some content can be dynamically added (I was thinking of the search). It lets the click events bubble up to the document level and then checks if the target qualifies, which is usually preferred when there's many possible targets vs. attaching to all anchor tags individually.

Reply:
updated to `closest` matcher to also check for `#navigation` ancestor